pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -213,4 +214,17 @@ def __add__(self, H):
 
         return NotImplemented
 
+    def __mul__(self, other):
+        ops = self.ops.copy()
+        coeffs_fixed = self.coeffs_fixed.copy()
+        coeffs_parametrized = self.coeffs_parametrized.copy()
+        if isinstance(other, TensorLike) and qml.math.ndim(other) == 0:
+            coeffs_fixed = [other * c for c in coeffs_fixed]
+            coeffs_parametrized = [
+                lambda p, t, new_c=c: other * new_c(p, t) for c in coeffs_parametrized
```

Comment:
Could this just be 
```suggestion
                lambda p, t  : other * c(p, t) for c in coeffs_parametrized
```
or is there a reason that would not work? Maybe a small detail, but I think its preferable when you look at `R.driving_interaction.coeffs_parametrized` that the functions are still just functions of `(p, t)` and not `(p, t, new_c=<function f1 at 0x14f253e20>)`. Maybe no one will do this, but I certainly look at it when things aren't working to make sure everything still looks the way I expect it.

Reply:
same thing with R.coeffs, which I think people are more likely to look at

Reply:
Your solution won't work, because `c` is defined in a loop, and when evaluating the `lambda` functions, `c` will correspond to the last value of the loop (all `lambda` functions will use the same `c`)

Reply:
Adding a `kwarg` is a nice solution to this problem, because now `c` is defined during the **definition** of the function (which happens inside the loop), and not during **evaluation**.

Reply:
Hahaha, right, I should have know that 😅